### PR TITLE
Add api tool call for testing

### PR DESCRIPTION
**Pull request by mosuem**
Check for breaking changes.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: .github/workflows/health_internal.yaml [hunk 1]
```diff
@@ -11,3 +11,5 @@ jobs:
     with:
       local_debug: true
       coverage_web: false
+      upload_coverage: false
+      checks: breaking
```
Comment (mosuem) [inline .github/workflows/health_internal.yaml:15]: To be reset before merge - this is just for faster debug iteration.

File: pkgs/firehose/lib/src/health/health.dart [hunk 2]
```diff
@@ -92,6 +99,43 @@ Documentation at https://github.com/dart-lang/ecosystem/wiki/Publishing-automati
...
+          'diff',
+          '--old',
+          basePackage,
+          '--new',
+          '.',
+        ],
+        workingDirectory: currentPath,
+      );
+      print('runApiTool: err:${runApiTool.stderr}, out:${runApiTool.stdout}');
+      totalOut += runApiTool.stdout.toString();
```
Comment (mosuem) [inline pkgs/firehose/lib/src/health/health.dart:127]: switch to json/markdown and format it nicely.

File: pkgs/firehose/lib/firehose.dart [hunk 3]
```diff
@@ -24,7 +24,9 @@ class Firehose {
   final Directory directory;
   final bool useFlutter;
 
-  Firehose(this.directory, this.useFlutter);
+  final String testArg;
```
Comment (mosuem) [inline pkgs/firehose/lib/firehose.dart:27]: This calls for a `DO_NOT_SUBMIT` check :)

File: pkgs/firehose/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 0.3.31
```
Comment (devoncarew) [inline pkgs/firehose/CHANGELOG.md:1]: Just fyi there are a few PRs underway rev'ing firehose -

File: .github/workflows/health.yaml [hunk 5]
```diff
@@ -92,6 +92,9 @@ jobs:
         run: dart pub global activate firehose
         if:  ${{ !inputs.local_debug }}
 
+      - name: Install api_tool
+        run: dart pub global activate --source git https://github.com/bmw-tech/dart_apitool
```
Comment (devoncarew) [inline .github/workflows/health.yaml:96]: I take it that this tool isn't published to pub?

File: pkgs/firehose/lib/src/health/health.dart [hunk 6]
```diff
@@ -92,6 +101,74 @@ Documentation at https://github.com/dart-lang/ecosystem/wiki/Publishing-automati
...
+    for (var package in packages) {
+      var currentPath =
+          path.relative(package.directory.path, from: Directory.current.path);
+      var basePackage = path.relative(
+        path.join(baseDirectory.absolute.path, currentPath),
+        from: currentPath,
+      );
+      print('Look for changes in $currentPath with base $basePackage');
+      var runApiTool = Process.runSync(
+        'dart-apitool',
```
Comment (devoncarew) [inline pkgs/firehose/lib/src/health/health.dart:118]: Perhaps instead `'dart, ['pub', 'global', 'run', 'dart-apitool', ...`? It's the same as what you have, just doesn't rely on the pub tools dir being installed into the PATH ...

File: pkgs/firehose/lib/src/health/health.dart [hunk 7]
```diff
@@ -92,6 +101,74 @@ Documentation at https://github.com/dart-lang/ecosystem/wiki/Publishing-automati
...
+        [
+          'diff',
+          ...['--old', basePackage],
+          ...['--new', '.'],
+          ...['--report-format', 'json'],
```
Comment (devoncarew) [inline pkgs/firehose/lib/src/health/health.dart:123]: If this run finds breaking changes, should we re-run this report in another format and dump the results to stdout?

**Review by devoncarew — APPROVED** [review]

**Pull request merged**